uni/progs/jsvalid.icn

```diff
@@ -0,0 +1,62 @@
+#
+############################################################################
+#
+#      File: jsvalid.icn 
+#    
+#
+#      Subject: This program takes two JSON files (one JSON object file, and one JSON Schema file)
+#      and compares the two files to make sure the JSON Object File conforms to the JSON Schema file. 
+#		  		
+#		
+#      Author: Hunter Thomas, John Haag, and Zachary Kovalenko
+#
+#      Date: April 2021     
+#
+############################################################################
+
+link basename
+link options
+link io
+link ximage
+
+import json
+
+
+procedure usage()
+    prog := basename(&progname)
+    write("Usage: ", prog, " [options]\n")
+    write(prog, " is a unicon utility for validating a json file given its schema")
+    write("Options:")
+    write("\t -f : json filename")
+    write("\t -s : json schema file, default to <fnamet>.schema.json")
+    write("\t -h : show this help\n")
+    exit(-1)
+end
+
+procedure main(args)
+   local opts, jfname, sfname, ext
+   # : means that an input is specified
+   opts := options(args, "hs:f:")
```

Comment:
The error messages from `options()` do not include any clues about which options are available. I think it would be better to include a custom procedure that, at least, reminds the caller of the -h option to get help.

Reply:
Not sure I follow. we do have `usage()` and it is called on various errors and the user can still to `-h` too.

Reply:
The point I'm making is that without a third argument to `options() `if, for example, the user types `jsvalid -help` they will get the standard error message `Unrecognized option: -e` with no clue as to what the valid options are. They have to "know" that it's `-h` or no parameters at all to print the help.  See the `usage() `proc in `uni/progs/idxgen.icn` for an example of what I'm proposing (although it perhaps doesn't need to be as elaborate as that; something that listed a summary of the options after the error message would be fine).
If you don't want to do that, that's fine too but (in my view) it makes it slightly harder to use.

Reply:
ah, well that is  easy, nothing but the best for you @Don-Ward  :)